### PR TITLE
Discovery server fixes <2.0.x> [10649]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Merge after #1793

This PR backports the following PRs to <2.0.x>

* Set client known as remote to local if is pinging server: #1751
* New Discovery-Server SUPER_CLIENT participant type: #1763
* Count new entities: #1650

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
Discovery Server test results:

	  7 - discovery_server_test.test_03_single_server_large.SHM_ON (Failed)
	  8 - discovery_server_test.test_03_single_server_large.SHM_OFF (Failed)
	 47 - discovery_server_test.test_23_fast_discovery_server_tool.SHM_ON (Failed)
	 48 - discovery_server_test.test_23_fast_discovery_server_tool.SHM_OFF (Failed)
	 53 - discovery_server_test.test_26_backup_restore.SHM_ON (Failed)

Test 03 (single server large) fails because the terminal has to be correctly configured.
Test 23 (fast discovery server tool) fails because #1392 has not been backported because it breaks previous behavior in ROS 2 Foxy
Test 26 (backup restore) seems to be flaky.

New tests associated with the backported PRs have passed.

**Comment by JLBuenoLopez-eProsima** [comment]
CI failures seem unrelated

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test discovery-server

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test discovery-server

File: include/fastdds/rtps/common/GuidPrefix_t.hpp [hunk 1]
```diff
@@ -93,12 +94,15 @@ inline std::ostream& operator <<(
...
     for (uint8_t i = 0; i < 11; ++i)
     {
-        output << (int)guiP.value[i] << ".";
+        output << std::setw(2) << (int)guiP.value[i] << ".";
     }
-    output << (int)guiP.value[11];
+    output << std::setw(2) << (int)guiP.value[11];
     return output << std::dec;
+    output.fill(old_fill);
+
```
Comment (EduPonz) [inline include/fastdds/rtps/common/GuidPrefix_t.hpp:105]:
```suggestion
```

**Review by EduPonz — DISMISSED** [review]
Only a style thing

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test discovery-server

**Review by EduPonz — APPROVED** [review]
LGTM

**Pull request merged**